storage/remote/otlptranslator/prometheusremotewrite/metrics_to_prw.go

```diff
@@ -260,3 +272,43 @@ func (c *PrometheusConverter) addSample(sample *prompb.Sample, lbls []prompb.Lab
 	ts.Samples = append(ts.Samples, *sample)
 	return ts
 }
+
+func NewResourceAttributesSetting(otlpCfg config.OTLPConfig) ResourceAttributesSetting {
+	if otlpCfg.IgnoreResourceAttributes == nil {
+		attr := make(map[string]struct{}, len(otlpCfg.PromoteResourceAttributes))
+		for _, s := range otlpCfg.PromoteResourceAttributes {
+			attr[s] = struct{}{}
+		}
+		return ResourceAttributesSetting{
+			Action: PromoteResourceAttributeAction,
+			Attr:   attr,
+		}
+	}
+	attr := make(map[string]struct{}, len(otlpCfg.IgnoreResourceAttributes))
+	for _, s := range otlpCfg.IgnoreResourceAttributes {
+		attr[s] = struct{}{}
+	}
+	return ResourceAttributesSetting{
+		Action: IgnoreResourceAttributeAction,
+		Attr:   attr,
+	}
+}
+
+func (s *ResourceAttributesSetting) isPromote(name string) bool {
```

Comment:
```suggestion
func (s *ResourceAttributesSetting) shouldPromote(name string) bool {
```

How does that sound to you? Maybe a bit clearer to read?